### ISSUE TITLE
Link to "Online Version" of a Communication Does Not Work

### ISSUE THREAD

**Issue by JimMichael** [SparkDevNetwork/Rock#4486]
### Description

When inserting a link back to a Communication for viewing by non-HTML clients (or just because you want to provide a "view on web" option in your emails), it shows an error instead of letting you see the communication.

### Steps to Reproduce

1. The link I'm referring to can be found under `Admin | Communications | Communication Mediums | Email` in the _Default Plain Text_ box. 

The text reads: _You can view an online version of this email here: _ and the link is `{{ 'Global' | Attribute:'PublicApplicationRoot' }}GetCommunication.ashx?c={{ Communication.Guid }}&p={{ Person | PersonActionIdentifier:'Unsubscribe' }}`

2. Copy that link into a new communication and send it to a person/address you can retrieve email for.

3. Click the link in the received communication and observe the error _Sorry, the communication you requested does not exist, or you are not authorized to view it._

**Expected behavior:**

I  would see a webpage with the HTML version of the communication when you click the link.

**Actual behavior:**

I get an error. I was not able to dupe this on demo as comms are disabled, but one person in chat reports the same behavior. I was able to observe it on both our 10.2 prod and 11.3 test instances here. The `PersonActionIdentifier:'Unsubscribe'` seems curious in this context, but that's what's there by default on core.

### Versions

* **Rock Version:** 10.2+

**Comment by cabal95** [comment]
I was able to pull up a few recent e-mails on our server and none of the 'view online' links worked for me. Which is odd. I didn't see anything obvious in the code as to why it would give the "not authorized" response it does. Generally that would only happen if the communication wasn't found or the person wasn't found - but it is supposed to fallback to the currently logged in person which I was logged in already.

**Comment by JimMichael** [comment]
It looks like maybe the PersonActionIdentifier shouldn't even be used for this link. Per the docs https://community.rockrms.com/lava/filters/person-filters#personactionidentifier only certain blocks know what to do with those tokens, and the communication page is not listed as one of them.

I know in prior versions of Rock it was a URLEncodedKey, so maybe switching to PersonActionIdentifier (which IS needed for the Unsubscribe link which is defined above this plain-text client link) was an oversight?

**Comment by nairdo** [comment]
@JimMichael  I did confirm that the GetCommunication.ashx (handler) is one of the items that can decode the 'Unsubscribe' identifier and I updated the Lava docs to note that.  We're investigating what might be happening in this issue.

**Comment by JimMichael** [comment]
Ok, at least in my case, this was all user error: 
1. It actually IS working on my 11.3 test instance (not sure why I thought it wasn't when I created this issue).
2. The code/link to use the *guid* of a communication instead *CommunicationId* was changed in 10.3, but somehow on my *10.2* production instance I had copied the NEW link that contained the Guid into my Email Medium ... which of course cannot work on 10.2. <insert facepalm> I probably copied the new link when I saw a user in chat post it, and saw that mine was "outdated"... sigh)

So from my perspective, this can be closed. It would be interesting to know if @cabal95 was also using "guid" link on < 10.3, though.
